time-it-builder/src/lib.rs

```diff
@@ -37,37 +77,123 @@ impl TimeItBuilder {
     /// This function is called when the timer is done.
     /// The first argument is the name of the timer, the second is the duration.
     pub fn time_it(&self, action: TimeItCallBack) -> &Self {
-        let mut config = TIMER_CONFIG.write();
+        let mut config = TIMEIT_CONFIG.write();
         config.action = action;
         self
     }
 
     pub fn default_tag(&self, tag: impl ToString) -> &Self {
-        let mut config = TIMER_CONFIG.write();
+        let mut config = TIMEIT_CONFIG.write();
         config.default_tag = tag.to_string();
         self
     }
+
+    pub fn mode(&self, mode: TimeItMode) -> &Self {
+        let mut config = TIMEIT_CONFIG.write();
+        config.mode = mode;
+        self
+    }
 }
 
 pub fn get_config() -> &'static TimeItConfig {
-    LAZY_TIMER_CONFIG.get_or_create(|| TIMER_CONFIG.read().clone())
+    LAZY_TIMEIT_CONFIG.get_or_create(|| {
+        let mut config = TIMEIT_CONFIG.write();
+        let mut lazy_config = std::mem::replace(&mut *config, TimeItConfig::default());
+        let default_tag = lazy_config.default_tag.clone();
+        let shutdown = config.shutdown.clone();
+        match lazy_config.mode {
+            TimeItMode::Async => {
+                let (sender, receiver) = bounded::<TimeItEvent>(100);
+                let handle = thread::spawn(move || {
+                    while !(shutdown.load(Ordering::Relaxed) && receiver.is_empty()) {
+                        match receiver.recv() {
+                            Ok(event) => {
+                                (lazy_config.action)(
+                                    event.tag.as_ref().unwrap_or(&default_tag),
+                                    event.duration,
+                                );
+                            }
+                            Err(_) => {
+                                break;
+                            }
+                        }
+                    }
+                });
+                lazy_config.sender = Some(sender);
+                // Keep the config in the original struct because we want to get mutable reference to it later for cleaning up
+                config.receiver_handle = Some(handle);
+                lazy_config
+            }
+            TimeItMode::Sync => lazy_config,
```

Comment:
as this gets more complex we should consider breaking this into its own init function instead of a closure

Reply:
makes sense